libraries/shared/storage/backfill/storage_value_loader.go

```diff
@@ -32,18 +32,24 @@ func NewStorageValueLoader(bc core.BlockChain, db *postgres.DB, initializers []s
 	}
 }
 
+type storageKey = common.Hash
+type storageValue = common.Hash
+type chunksOfKeysToValues = []map[storageKey]storageValue
+
 type StorageValueLoader struct {
-	bc              core.BlockChain
-	db              *postgres.DB
-	HeaderRepo      datastore.HeaderRepository
-	StorageDiffRepo storage2.DiffRepository
-	initializers    []storage.TransformerInitializer
-	startingBlock   int64
-	endingBlock     int64
+	bc                      core.BlockChain
+	db                      *postgres.DB
+	HeaderRepo              datastore.HeaderRepository
+	StorageDiffRepo         storage2.DiffRepository
+	addressesToKeysToValues map[common.Address]chunksOfKeysToValues
+	initializers            []storage.TransformerInitializer
+	startingBlock           int64
+	endingBlock             int64
 }
 
 func (r *StorageValueLoader) Run() error {
-	addressToKeys, getKeysErr := r.getStorageKeys()
+	r.addressesToKeysToValues = make(map[common.Address]chunksOfKeysToValues, len(r.initializers))
```

Comment:
It's weird that you're initializing this here, but not filling it until `setStorageKeys`. I realize `setStorageKeys` is private, but it's still..off.

Reply:
You could initialize this field in the `NewStorageValueLoader` - since `initializers` isn't changed except on construction you can also make the `map`.

Reply:
Good call!

Reply:
The only dilemma this creates is that if we do this, you can't initialize an instance of the `StorageValueLoader` with a literal (as we do in the tests). Normally my move here would be to make the `struct` private and define a public interface so that you have to use a constructor to get an implementation, but that would also break existing tests - which depend on being able to modify public properties on the `struct`. Though maybe it makes sense to change the constructor to take all the properties that need to be mocked in the tests as args? 🤔